src/HttpAsyncClientDiscovery.php

```diff
@@ -24,7 +24,7 @@ public static function find()
         try {
             $asyncClient = static::findOneByType(HttpAsyncClient::class);
 
-            return new $asyncClient();
+            return static::instantiateClass($asyncClient);
```

Comment:
Correct me if I am wrong, but this try-catch only handles DiscoveryFailedExceptions, which mean we cannot (and we don't want to) catch VeryRandomExceptions.